common/src/main/java/com/tencent/rss/common/util/RssUtils.java

```diff
@@ -267,4 +267,14 @@ public static void checkQuorumSetting(int replica, int replicaWrite, int replica
       throw new RuntimeException("Replica config is unsafe, recommend replica.write + replica.read > replica");
     }
   }
+
+  // the method is used to transfer hostname to metric name.
+  // With standard of hostname, `-` and `.` will be included in hostname,
+  // but they are not valid for metric name, replace them with '_'
+  public static String getMetricNameForHostName(String hostName) {
+    if (hostName == null) {
+      return "";
+    }
+    return hostName.replaceAll("[\\.-]", "_");
```

Comment:
What about ' '(i.e., blank)?

Reply:
For hostname, standards(eg, RFC 1035) which only contains `[a-z], -, .` are applied, the other character is not allowed. If the rule is broken, update the method according to new rule.

Reply:
That is OK.
LT2M.